content/en/blog/_posts/2022-01-24-POV-Kubernetes-cluster-infrastructure-upgrade-strategies.md

```diff
@@ -0,0 +1,94 @@
+## Context
+
+This blog is an outcome of the experience gained while working for a client having a Kubernetes based platform hosted on infrastructure (EC2 instances) hosted on AWS. AWS releases new images to be used to build the servers (EC2 instances) every now and then, when a new vulnerability fix is made, a newer feature is added etc. This client had additional hardening to be done on those images and a custom image is released internally for the entire organization to use. There was a deadline imposed by when old the images must be stopped to be in use and the infrastructure is to be built using only the latest image available. One of the teams that is impacted by this was the platform team, the team that hosts and manages a shared platform based on Kubernetes. Many applications teams in the organization host their applications on this shared platform as independent containers. 
+Every time a new image is available, all the servers that are a part of the Kubernetes cluster were to be upgraded.  The Blue green strategy mentioned in this paper was one of the strategies followed. All the applications using the shared platform were redeployed when a new cluster using the latest EC2 images is built. There was a lead time involved in validating all the applications redeployed and all the application stakeholders are involved for this validation exercise. The more the applications, the more the lead time.
+There was hence a need to explore any other possible options to reduce the dependency on the application teams to validate each time the applications are deployed into a newly built cluster.
+
+## Need for the Infrastructure management
+
+Managing a platform includes the activity of ensuring that the underlying infrastructure is maintained, upgraded, and patched at regular intervals as applicable to take care of latest features, capabilities or vulnerability fixes that are available. Handling these activities with minimal or zero impact to the business is crucial.
+Platform agnostic and loosely coupled code base, Scalability, Reliability, Self-healing capability are the current generation basic requirements for the development, roll out and maintenance of applications for any digital customer to be able to keep up with the ever-changing demands, technologies, tools, and business strategies. Kubernetes is one such an open-source container orchestration platform that allows customers to run their applications effectively as containers without worrying about the scale of the application, the underlying OS, hardware, or any other interdependencies. 
```

Comment:
Capitalization & grammatical nit
```suggestion
Platform agnostic and loosely coupled code base, scalability, reliability, self-healing capability are the current generation basic requirements for the development, roll out and maintenance of applications for any digital customer to be able to keep up with the ever-changing demands, technologies, tools, and business strategies. Kubernetes is one such open-source container orchestration platform that allows customers to run their applications effectively as containers without worrying about the scale of the application, the underlying OS, hardware, or any other interdependencies. 
```

Reply:
Platform agnostic and loosely coupled code base, scalability, reliability, self-healing capability are the current generation basic requirements for the development, roll out and maintenance of applications for any digital customer to be able to keep up with the ever-changing demands, technologies, tools, and business strategies - Do we intend to use generation? Or please can you clarify what is being communicated here?